### PR TITLE
New package: zcash-1.0.4

### PR DESCRIPTION
**Pull request by ananteris**

### CODE REVIEW

File: srcpkgs/zcash/template [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'zcash'
+pkgname=zcash
+version=1.0.4
+revision=1
+makedepends="glibc boost"
```
Comment (Vaelatern) [inline srcpkgs/zcash/template]: Never depend on a libc.

File: srcpkgs/zcash/template [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+hostmakedepends="zlib-devel libgomp-devel wget autoconf perl m4 autoconf automake libtool pkg-config python"
```
Comment (Vaelatern) [inline srcpkgs/zcash/template]: PLEASE read the Manual sections about `hostmakedepends` vs. `makdepends`.

**Comment by ananteris** [comment]
m4 has been added to hostmakedepends. Are there any other outstanding issues?

**Comment by Duncaen** [comment]
This downloads a whole bunch of dependencies at compile time and builds them, can we use the system wide ones instead?

**Comment by ananteris** [comment]
No, you cannot. Removing any single one will break the install. That is why they are in there. Curiously enough with the most recent version of void-packages even more hostdepends (autoconf, m4, automake, libtool) are needed.

**Comment by Vaelatern** [comment]
Does the build try to use its own tools? Or does it use ours?

**Comment by ananteris** [comment]
It uses exactly the tools the developers of zcash intended for it to. I am not aware of any script that is pulling its own version of m4/automake/autoconf or libtool but if it did there would likely be a reason for it. At the same time, this are listed as hostmakedepends for literally thousands of other programs in the repo (in the case of automake), why is my package held to a different standard than any of them?

**Comment by Duncaen** [comment]
> At the same time, there are literally thousands of other programs in the repo (in the case of automake) that list these as hostmakedepends. Why is my package held to a different standard than any of them?

I don't see the problem, what do you mean by different standards?

Can you at least disable the ccache that it downloads?
And maybe instead of fetching the archives at build time, provide them as `distfiles` to be able to cache them?

**Comment by Vaelatern** [comment]
This is now obsolete, several newer versions of zcash have been released.

**Pull request closed without merging**